packages/flutter/lib/src/material/slider_theme.dart

```diff
@@ -0,0 +1,729 @@
+///  * The [Slider] is disabled when it is not accepting user input. See
+///    [Slider] for details on when this happens.
+///
+/// The thumb and the value indicator may have their shapes and behavior
+/// customized by creating your own [SliderComponentShape] that does what
+/// you want. See [RoundSliderThumbShape] and
+/// [PaddleSliderValueIndicatorShape] for examples.
+///
+/// See also:
+///
+///  * [SliderTheme] widget, which can override the slider theme of its
+///    children.
+///  * [Theme] widget, which performs a similar function to [SliderTheme],
+///    but for overall themes.
+///  * [ThemeData], which has a default [SliderThemeData].
+///  * [SliderComponentShape], to define custom slider component shapes.
+class SliderThemeData extends Diagnosticable {
+  /// Create a [SliderThemeData] given a set of exact values. All the values
+  /// must be specified.
+  ///
+  /// This will rarely be used directly. It is used by [lerp] to
+  /// create intermediate themes based on two themes.
+  ///
+  /// The simplest way to create a SliderThemeData is to use
+  /// [copyWith] on the one you get from [SliderTheme.of], or create an
+  /// entirely new one with [SliderThemeData.materialDefaults].
+  const SliderThemeData({
+    @required this.activeRailColor,
+    @required this.inactiveRailColor,
+    @required this.disabledActiveRailColor,
+    @required this.disabledInactiveRailColor,
+    @required this.activeTickMarkColor,
+    @required this.inactiveTickMarkColor,
+    @required this.disabledActiveTickMarkColor,
+    @required this.disabledInactiveTickMarkColor,
+    @required this.thumbColor,
+    @required this.disabledThumbColor,
+    @required this.overlayColor,
+    @required this.valueIndicatorColor,
+    @required this.thumbShape,
+    @required this.valueIndicatorShape,
+    @required this.showValueIndicator,
+  }) : assert(activeRailColor != null),
+       assert(inactiveRailColor != null),
+       assert(disabledActiveRailColor != null),
+       assert(disabledInactiveRailColor != null),
+       assert(activeTickMarkColor != null),
+       assert(inactiveTickMarkColor != null),
+       assert(disabledActiveTickMarkColor != null),
+       assert(disabledInactiveTickMarkColor != null),
+       assert(thumbColor != null),
+       assert(disabledThumbColor != null),
+       assert(overlayColor != null),
+       assert(valueIndicatorColor != null),
+       assert(thumbShape != null),
+       assert(valueIndicatorShape != null),
+       assert(showValueIndicator != null);
+
+  /// Generates a SliderThemeData from three main colors.
+  ///
+  /// Usually these are the primary, dark and light colors from
+  /// a [ThemeData].
+  ///
+  /// The opacities of these colors will be overridden with the Material Design
+  /// defaults when assigning them to the slider theme component colors.
+  ///
+  /// This is used to generate the default slider theme for a [ThemeData].
+  factory SliderThemeData.materialDefaults({
```

Comment:
i would probably call this `SliderThemeData.fromPrimaryColors`. The name `SliderThemeData.materialDefaults` sounds like a constant that has material defaults, not a method that computes values from other values.

Reply:
Hmm. Ok, but the reason I called it that was because the opacities that it overrides the given colors with, and the shapes it assigns for the components are all from the Material spec.  This is also the method used to create the default theme for the Material ThemeData.

Reply:
being from the material spec is implied, this is the material library. :-)